apollo-router/src/query_planner/mod.rs

```diff
@@ -297,19 +300,24 @@ impl QueryPlan {
     }
 }
 
+// holds the query plan executon arguments that do not change between calls
+pub(crate) struct ExecutionParameters<'a, SF> {
+    context: &'a Context,
+    service_factory: &'a Arc<SF>,
+    schema: &'a Schema,
+    originating_request: &'a Arc<http_ext::Request<Request>>,
+    deferred_fetches: &'a HashMap<String, Sender<(Value, Vec<Error>)>>,
+    options: &'a QueryPlanOptions,
+}
+
 impl PlanNode {
     #[allow(clippy::too_many_arguments)]
```

Comment:
we can probably remove that now

Reply:
Remove what?

Reply:
I just removed `     #[allow(clippy::too_many_arguments)] ` which isnt required anymore :D